x-pack/plugins/upgrade_assistant/public/application/components/tabs/checkup/deprecations/reindex/flyout/warning_step_checkbox.tsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import {
+  EuiCheckbox,
+  EuiCode,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIconTip,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { DocLinksStart } from 'kibana/public';
+import { ReindexWarning, ReindexWarningTypes } from '../../../../../../../../common/types';
+
+export const hasReindexWarning = (
+  warnings: ReindexWarning[],
+  warningType: ReindexWarningTypes
+): boolean => {
+  return Boolean(warnings.find((warning) => warning.warningType === warningType));
+};
+
+const WarningCheckbox: React.FunctionComponent<{
+  isChecked: boolean;
+  warningId: string;
+  label: React.ReactNode;
+  description: React.ReactNode;
+  documentationUrl: string;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}> = ({ isChecked, warningId, label, onChange, description, documentationUrl }) => (
+  <>
+    <EuiText>
+      <EuiFlexGroup justifyContent="spaceBetween">
+        <EuiFlexItem grow={false}>
+          <EuiCheckbox
+            id={warningId}
+            label={<strong>{label}</strong>}
+            checked={isChecked}
+            onChange={onChange}
+          />
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiLink href={documentationUrl} target="_blank" external={false}>
+            <EuiIconTip
+              content={
+                <FormattedMessage
+                  id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.documentationLinkLabel"
+                  defaultMessage="Documentation"
+                />
+              }
+              position="right"
+              type="help"
+            />
+          </EuiLink>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      <EuiSpacer size="xs" />
+
+      {description}
+    </EuiText>
+
+    <EuiSpacer />
+  </>
+);
+
+export interface WarningCheckboxProps {
+  isChecked: boolean;
+  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  docLinks: DocLinksStart['links'];
+  id: string;
+  meta?: ReindexWarning['meta'];
+}
+
+export const CustomTypeNameWarningCheckbox: React.FunctionComponent<WarningCheckboxProps> = ({
+  isChecked,
+  onChange,
+  docLinks,
+  id,
+  meta,
+}) => {
+  return (
+    <WarningCheckbox
+      isChecked={isChecked}
+      onChange={onChange}
+      warningId={id}
+      label={
+        <FormattedMessage
+          id="xpack.upgradeAssistant.checkupTab.reindexing.flyout.warningsStep.customTypeNameWarningTitle"
+          defaultMessage="Mapping type {mappingType} will be changed to {defaultType}"
```

Comment:
This message & description are a bit contradictory--if mapping types aren't supported, why is the mapping type being renamed? I think fewer words might be better here.

```suggestion
          defaultMessage="Replace mapping type {mappingType} with {defaultType}"
```